### DISCUSSION TITLE
Creating new form in Open MCT and Login form in open mct

### DISCUSSION THREAD

**Discussion by Vaibhav2121** [nasa/openmct#3809] [Community Q&A]
Can we create a login form and a form in open mct where user can select parameters to watch in index.html, first page of openmct.

**Comment by jvigliotta** [comment]
Hi @Vaibhav2121, we don't have login form development coming up in any of our development plans as far as I know, but it's possible someone in the community has done something similar. As far as the parameters to watch in index.html, I'm not too sure what you mean by that. Some plugins accept configuration objects/values, our time conductor is a good example of that. It would be possible to do that dynamically as well to update those items.

**Comment by davetsay** [comment]
@davetsay

**Comment by davetsay** [comment]
@Vaibhav2121 , you can create forms using the [Forms API](https://github.com/nasa/openmct/blob/master/src/api/forms/FormsAPI.js). A very simple form example is below, but you can customize and even extend forms for your use case. Open MCT also has a [User API](https://github.com/nasa/openmct/blob/master/API.md#user-api).

I am now understanding your use case though, to use Open MCT as an adhoc monitoring tool. I still think you should strive for the answer provided in #3806, but you could create a custom plugin that would accomplish what you desire. Take a look at [Form Location Selector](https://github.com/nasa/openmct/blob/5872b9006d8b6d3eef0b18007ad691218a4bea34/src/plugins/formActions/CreateWizard.js#L107-L136), which provides a means to select one parameter, a location to save in this case, from the dictionary.

@jvigliotta is working with a group that is creating a modified table view in which parameters are subscribed to in an adhoc fashion via a selection tool. He might be able to speak more to that if indeed that is what you are trying to do.

```
const formStructure = {
  title: 'My Login Form',
  sections: [{
    name: 'Login',
    cssClass: 'grows',
    rows: [
      {
        key: 'username',
        name: 'User Name',
        cssClass: 'l-input-lg',
        control: 'textfield',
        required: true,
      },
      {
        key: 'password',
        name: 'Password',
        cssClass: 'l-input-lg',
        control: 'textfield',
        required: true,
        validate: (data) => {
          // validate password entry
        }
      }
    ]
  }]
};

openmct.forms.showForm(formStructure).then((data) => {
  // handle form data submission
});
```
